testapps/test_pillowtop/tests/test_xform_pillow.py

```diff
@@ -69,6 +77,19 @@ def test_form_soft_deletion(self):
         results = FormES().run()
         self.assertEqual(0, results.total)
 
+    @run_with_all_backends
+    def test_form_pillow_error_in_form_metadata(self):
+        self.assertEqual(0, PillowError.objects.filter(pillow='xform-pillow').count())
+        with patch('pillowtop.processors.form.mark_latest_submission') as mark_latest_submission:
+            mark_latest_submission.side_effect = ResourceConflict('couch sucks')
+            case_id, case_name = self._create_form_and_sync_to_es()
+
+        # confirm change made it to form index
+        results = FormES().run()
+        self.assertEqual(1, results.total)
```

Comment:
this is purely dependent on the ordering of the processors correct?

Reply:
Kind of. But because the mock is mocking a function only used in a different processor, it should be robust against changes we make to handle failures with more granularity. In other words, if we tried following processors even if one in the middle failed, this test should still pass.

Is there a change you're considering that would cause this test to fail?

Reply:
not suggesting any changes, was just wanting to understand. I thought that if an earlier processor failed then later ones would not be called but I see it's different for batch and serial:
**batch**
- Try batch for next processor in list
- if processor raises exception try each change serially for that processor
- if any change fails create an error record
- move to next processor

**serial**
- run each processor for each change
- if any processor fails create error record and move to next change (later processors are skipped for that change)

This means that we could end up with multiple error records per change in the batch processing case (just making a note)

Reply:
> This means that we could end up with multiple error records per change 

There's a unique index for doc_id and pillow so that shouldn't be possible https://github.com/dimagi/commcare-hq/blob/868148af1ef1c32dcadfca265866ad4bd53fa1e8/corehq/ex-submodules/pillow_retry/models.py#L56